### ISSUE TITLE
External authentication really broken/buggy?

### ISSUE THREAD

**Issue by LindsayL** [edegraff/FundTracker#5]
First:
Tried to log in with google, when I was asked to submit my username I left it is my e-mail ***_@**__._*
Gave this error:

![image](https://cloud.githubusercontent.com/assets/5545461/6499251/15c92016-c2b5-11e4-9b6e-cd63416ea989.png)

Then I tried just a lowercase string, same result.

Gave up, logged in with a non-external account, went to manage account, hit google under 'add external login', it added the account I was previously trying to log into google with!  (Big security flaw)

Clicking google again did nothing, and hitting remove followed by google again results in the account being automatically added again.  (Buggy and security flaw?)

Logged out went to try to log in with an external account again.  It took me straight to the select username page where, it indicated that my username was already in use.  Entering a new username causes it to crash again.

**Comment by LindsayL** [comment]
I didn't look into too much but when I examined the database, AzureConnection->UserProfile:
-there was an entry for each user name that I entered that caused a crash.
-there was no entry for my non-external user account?  (Where should I be looking?)

**Comment by StevenGerdes** [comment]
We have to change logging in anyways because we wanted to remove username and have only email. I quickly looked into the issue and we don't save external anything except username, also it should say "username in use" instead of crashing (I attempted to reproduce the issue I got a crash the first attempt and the second attempt it simply said "username was in use") also just fyi the screenshot doesn't have high enough resolution to read. I have two midterms Friday and will not be able to fix this until later. But feel free to do look into earlier if you want.
From using my two google accounts I can confirm that each time you sign in with an external account it returns (over https so it is safe ) a unqiue and repeatable user id. This id can be used as the password and the email as the username. I suggest we don't allow people to hook up external accounts after the fact at all. Either they create an account supplying us with their email/password or an external one. Then if they (for some reason) attempt to register with an existing email just say that it exists.

**Comment by LindsayL** [comment]
Sorry I guess temporarily holding the screen shot in MS word must've killed the resolution.

When your's crashed on the first attempt is almost for sure the same crash I was receiving.  It's strange, I know I tested all the login functionality earlier when I was working on the GUI and it all seemed to work.

Either way, I suppose it's not important unless it persists after the changes you listed.  I also just took a brief scan through our requirements and I didn't see any explicit references to being able to create an account without google authentication.  (In fact we mention creating an account with google twice in row [3.2.1.3.2 and 3.2.1.3.3] just slightly different wording, a typo?)  So we might be able to reduce it to just google authentication if we want.  (Though I think it's probably better if we leave the option in.  If we do, should we do one of those e-mail verifications? [Maybe only if we have time])

I was going to start working on the funds list page (displaying values, and then the details page), so if it's alright, can I leave this with you?
